app/src/main/java/de/westnordost/streetcomplete/map/components/TracksMapComponent.kt

```diff
@@ -22,46 +22,52 @@ class TracksMapComponent(ctrl: KtMapController) {
 
     private var index = 0
     private var tracks: MutableList<MutableList<LngLat>>
+    private var tracksRecording: MutableList<Boolean>
```

Comment:
I'd prefer something like this. Put into one data structure what belongs together.

```suggestion
    private class Track(val trackpoints: MutableList<LngLat>, isRecording: Boolean)

    private var tracks: MutableList<Track>
```

Reply:
I wanted to explicitly enforce that there is only one historical recording at a time. This is stored in the `tracksRecording`. There can be many previous segments of recording tracks, but only the most recent one should be uploaded. I guess with this data structure I would loop through and find the last one which has `isRecording` set to true and this is the track that is uploaded?

Reply:
I don't quite understand. With this data structure, you don't enforce that there is only one recording. E.g.
```kotlin
tracks = listOf(listOf(p1,p2,p3), listOf(p4,p5))
tracksRecording = listOf(true, true)
```
-> both are "isRecording==true" tracks.

Also, you write 
> There can be many previous segments of recording tracks, but only the most recent one should be uploaded

but the `TracksMapComponent` has nothing to do with upload. And it has also nothing to do with storing the track temporarily before upload. It exclusively displays the track(s), nothing else.

Reply:
You are completely right here, apologies.